src/main/java/com/getindata/connectors/http/internal/table/lookup/HttpLookupTableSource.java

```diff
@@ -94,21 +100,41 @@ public LookupRuntimeProvider getLookupRuntimeProvider(LookupContext lookupContex
         PollingClientFactory<RowData> pollingClientFactory =
             createPollingClientFactory(lookupQueryCreator, lookupConfig);
 
-        HttpTableLookupFunction dataLookupFunction =
-            new HttpTableLookupFunction(
-                pollingClientFactory,
-                responseSchemaDecoder,
-                lookupRow,
-                lookupConfig
-            );
+        // In line with the JDBC implementation and current requirements, we are only
+        // supporting Partial Caching for synchronous operations.
+        return getLookupRuntimeProvider(lookupRow, responseSchemaDecoder, pollingClientFactory);
+    }
 
+    protected LookupRuntimeProvider getLookupRuntimeProvider(LookupRow lookupRow,
+        DeserializationSchema<RowData> responseSchemaDecoder,
+        PollingClientFactory<RowData> pollingClientFactory) {
         if (lookupConfig.isUseAsync()) {
+            HttpTableLookupFunction dataLookupFunction =
+                    new HttpTableLookupFunction(
+                            pollingClientFactory,
+                            responseSchemaDecoder,
+                            lookupRow,
+                            lookupConfig
+                    );
             log.info("Using Async version of HttpLookupTable.");
             return AsyncTableFunctionProvider.of(
-                new AsyncHttpTableLookupFunction(dataLookupFunction));
+                    new AsyncHttpTableLookupFunction(dataLookupFunction));
```

Comment:
could you remind me why we could use Flink Cache for AsyncTableLookupFucntion?

Reply:
Do you mean `could` -> `couldn't `?

So my thinking was mostly pragmatic:
- JDBC only supports sync
- there are no implementations of async. 
- it looked like it would work - with the future using our thread pool executor, I was not sure if there would be 
2 future threads, one for the framework an one for us. I did not investigate further into this.
- I originally coded it using the AsyncTableLookupFunction and had junit failures. 
- for us we only need sync.


 

-